Sources/ApodiniDeploy/ApodiniDeployInterfaceExporter+ExportWebServiceStructure.swift

```diff
@@ -10,41 +10,28 @@ import Foundation
 import Apodini
 import ApodiniUtils
 import ApodiniOpenAPI
+import ApodiniDeployRuntimeSupport
 import ApodiniDeployBuildSupport
 @_implementationOnly import Vapor
 import OpenAPIKit
 
 
 extension ApodiniDeployInterfaceExporter {
-    func exportWebServiceStructure(to outputUrl: URL, apodiniDeployConfiguration: ApodiniDeploy.ExporterConfiguration) throws {
-        let deploymentConfig = apodiniDeployConfiguration.config
-        guard let openApiDocument = app.storage.get(OpenAPI.StorageKey.self)?.document else {
-            throw ApodiniDeployError(message: "Unable to get OpenAPI document")
+    func exportDeployedSystemIfNeeded() throws {
+        guard let structureExporter = app.storage[DeploymentStructureExporterStorageKey.self] else {
+            return
         }
-        var allDeploymentGroups: Set<DeploymentGroup> = deploymentConfig.deploymentGroups
+        var allDeploymentGroups: Set<DeploymentGroup> = self.exporterConfiguration.config.deploymentGroups
         allDeploymentGroups += explicitlyCreatedDeploymentGroups.map { groupId, handlerIds in
             DeploymentGroup(id: groupId, handlerTypes: [], handlerIds: handlerIds)
         }
-        let webServiceStructure = WebServiceStructure(
-            endpoints: Set(collectedEndpoints.map { endpointInfo -> ExportedEndpoint in
-                let endpoint = endpointInfo.endpoint
-                return ExportedEndpoint(
-                    handlerType: endpointInfo.handlerType,
-                    handlerId: endpoint[AnyHandlerIdentifier.self],
-                    deploymentOptions: endpointInfo.deploymentOptions,
-                    userInfo: [:]
-                )
-            }),
-            deploymentConfig: DeploymentConfig(
-                defaultGrouping: deploymentConfig.defaultGrouping,
-                deploymentGroups: allDeploymentGroups
-            ),
-            openApiDocument: openApiDocument,
-            enabledDeploymentProviders: apodiniDeployConfiguration.runtimes.map { $0.identifier }
-        )
-        try webServiceStructure.writeJSON(
-            to: outputUrl,
+        let config = DeploymentConfig(defaultGrouping: self.exporterConfiguration.config.defaultGrouping, deploymentGroups: allDeploymentGroups)
+        
+        let deployedSystem = try structureExporter.retrieveStructure(Set(self.collectedEndpoints), config: config, app: self.app)
+        try deployedSystem.writeJSON(
+            to: URL(fileURLWithPath: structureExporter.filePath),
             encoderOutputFormatting: [.prettyPrinted, .withoutEscapingSlashes]
         )
+        exit(EXIT_SUCCESS)
```

Comment:
As noted in my comment on the `WebService` type it might make sense to move the exit call in a structured and predicable way to the `WebService` start function if e.g. any other logic would have to be executed after the structure exporter and the server doesn't start up If e.g. a developer forgets to add the exit call to their structured exporter.

Reply:
Not sure if I understand your comments correctly, but there should be no need for a developer to add an exit call by themselves. This method `exportDeployedSystemIfNeeded()` is only executed if the web service is started by the `export-ws-structure [subcommand]` command. For this we don't need any other logic to be executed after the structure exporter has exported the system nor should a developer manually add exit calls anywhere. 
The developer does not have to create their own structure exporter, but just needs to implement `StructureExporter` where he can define how the system is exported and override `run` to start web service. 
But why would he need to pass here any flags how long the web service runs, if the only purpose of this is to export the structure anyway?
While I quite get the intention to provide some generic and nicer way to give a clearer semantic meaning than `waitForCompletion`, I don't think this should be in the scope of the refactoring and honestly I can't see any use cases for this at least for now. 
As said, maybe I misunderstood you, so please clarify if so 👍

Reply:
The intention behind the refactoring was/is the following: I think it makes is easier to reason about the behavior of the web service if we reduce or even completely eliminate the `exit(...)` calls across the code base and focus on a clear and followable startup + shutdown routine in the WebService. Your example has shown why this is needed. I should have been clearer what I have meant regarding a "developer" and the exit calls: E.g. some other Subcommand (e.g. database reverts, database migrations, generating a migration guide like Eldi is doing) will all have to manually put exit calls in their exporters or custom configuration logic.
Bundling this in the start command and therefore insuring that there are two (and could even be extended to more in the future) startup routines as detailed in my comments above would make this clearer for implementing subcommands that don't rely on starting up the web service completely.
Therefore they would call `try Service.start(options: .startOnly, app: app, webService: Service())` and all exporters and everything would be finished and done and the web service exits controlled after all the work without booting the web service and/or going in a run loop to process requests.

As this surfaces with this PR I would use the "see something, do something" approach and try to make it nice right awai instead of ending up with multiple dependencies implementing this themselves and having to introduce a change that they have to adapt to eventually later on.

Reply:
Thanks for clarifying!
I agree, this definitely makes sense. I will have a look at this. 👍

Reply:
@hendesi Yes, I think this reflects my vision. I had two comments regarding the changes that you can find below